### PR TITLE
feat: voice-memos interactive transcribe UX + system-wide locale dates

### PR DESCRIPTION
**Pull request by genesiscz**
## Summary

- **System-wide date formatting** (`src/utils/date.ts`): `formatDateTime()` with locale detection (macOS AppleLocale → env vars → Intl fallback), supporting absolute (9 formats), relative (4 modes), and combined display
- **Date migration**: All 13 hardcoded `"en-US"`/`"en-GB"` date formatting callsites migrated to `formatDateTime()`
- **Voice memos transcribe UX overhaul**:
  - `transcribe` without `<id>` shows memo selector instead of erroring
  - Language detection returns top-k probabilities via softmax over whisper logits
  - Interactive language confirmation prompt with alternatives
  - Prompts for provider/model/format/output when not given via CLI
  - Settings summary displayed before transcription starts
  - Suppress repeated HF "content-length" download warnings
  - Better error messages for empty `--provider`/`--model`
- **New reusable utilities**:
  - `src/utils/prompts/clack/settings-summary.ts` — formatted settings display
  - `src/utils/ai/suppress-warnings.ts` — targeted console.warn suppression
  - `src/macos/lib/voice-memos/prompts.ts` — shared DRY prompts

## Test plan

- [ ] `tools macos voice-memos` — interactive mode uses shared prompts
- [ ] `tools macos voice-memos transcribe` — shows memo selector + all prompts
- [ ] `tools macos voice-memos transcribe --lang cs` — shows memo selector, skips lang prompt
- [ ] `tools macos voice-memos transcribe 5 --provider local-hf --format srt -o out.srt` — no prompts
- [ ] `tools macos voice-memos transcribe --provider ""` — shows valid providers
- [ ] `tools macos voice-memos list` — dates in system locale
- [ ] Verify no remaining hardcoded `"en-US"` in CLI tools

## Summary by CodeRabbit

* **New Features**
  * Interactive voice-memo flows (select memo/action, model/provider, output)
  * File-path prompt with tab-completion and cancel symbol
  * Settings-summary renderer for CLI confirmations

* **Improvements**
  * Consistent, locale-aware date/time formatting across the app
  * Interactive language confirmation during transcription
  * Better model/provider discovery, validation, and cache detection
  * Hardened transcription error recovery and retry behavior
  * Exposed multiple language-detection alternatives

### CODE REVIEW

File: src/macos/commands/voice-memos/index.ts [hunk 1]
```diff
@@ -201,27 +213,159 @@ interface TranscribeOpts {
...
+    }
+
+    // Format — only prompt if not provided via CLI
+    if (opts.format && process.argv.includes("--format")) {
+        resolved.format = opts.format;
+    } else if (isTTY) {
+        resolved.format = await selectFormat();
+    } else {
+        resolved.format = opts.format ?? "text";
+    }
```
[the remark this answers is not in the document]
Comment (genesiscz) [reply src/macos/commands/voice-memos/index.ts]: Fixed in [595ad541](https://github.com/genesiscz/GenesisTools/commit/595ad5416b5bf7b1b791532f13c1ca9543e2b286) — removed process.argv.includes check entirely. Since the Commander default was already removed from the option definition, opts.format is only set when explicitly passed. /gemini

[the remark this answers is not in the document]
Comment (genesiscz) [reply src/macos/commands/voice-memos/index.ts]: Fixed in [595ad541](https://github.com/genesiscz/GenesisTools/commit/595ad5416b5bf7b1b791532f13c1ca9543e2b286) — simplified to just check opts.format (no Commander default means it is only set when explicitly passed). @coderabbitai

File: src/macos/lib/voice-memos/prompts.ts [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+    if (choice === "__other__") {
+        const custom = await p.text({
+            message: "Enter ISO language code (e.g. cs, en, de)",
+            placeholder: "cs",
+            validate: (val) => {
+                if (!val || val.length < 2) {
+                    return "Enter a valid 2-letter language code";
+                }
+
+                return undefined;
```
[the remark this answers is not in the document]
Comment (genesiscz) [reply src/macos/lib/voice-memos/prompts.ts:252]: Fixed in [595ad541](https://github.com/genesiscz/GenesisTools/commit/595ad5416b5bf7b1b791532f13c1ca9543e2b286) — tightened validation to /^[a-z]{2,3}$/i to accept ISO 639-1 (2 char) and ISO 639-3 (3 char) codes. @coderabbitai

File: src/utils/ai/ModelManager.ts [hunk 3]
```diff
@@ -6,7 +6,50 @@ import { formatBytes } from "@app/utils/format";
...
+        ...(process.env.GROQ_API_KEY
+            ? [
+                  { id: "whisper-large-v3-turbo", name: "Groq whisper-large-v3-turbo", description: "fast" },
+                  { id: "whisper-large-v3", name: "Groq whisper-large-v3", description: "high quality" },
+              ]
+            : []),
+        ...(process.env.OPENAI_API_KEY
+            ? [{ id: "whisper-1", name: "OpenAI whisper-1", description: "reliable" }]
+            : []),
+    ],
```
[the remark this answers is not in the document]
Comment (genesiscz) [reply src/utils/ai/ModelManager.ts]: Fixed in [595ad541](https://github.com/genesiscz/GenesisTools/commit/595ad5416b5bf7b1b791532f13c1ca9543e2b286) — moved cloud model list into a lazy getCloudTranscriptionModels() function that checks env vars at call time instead of module load time. @coderabbitai
